### PR TITLE
Move some of our test utilities to galsim.utilities

### PR DESCRIPTION
**Pull request by rmjarvis**
Some of the things that are in our galsim_test_helpers, I've used in other projects and just copied them over.  It would be easier if these lived in galsim.utilities instead, so I could just use them directly.  This PR puts the ones that I think are potentially useful for other project in our installed code base:

* do_pickle
* all_objs_diff
* timer
* profile
* CaptureLog

### CODE REVIEW

File: galsim/utilities.py [hunk 1]
```diff
@@ -1913,3 +1913,244 @@ def __exit__(self, type, value, traceback):
...
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.handler.flush()
+        self.output = self.stream.getvalue().strip()
+        self.handler.close()
+
+
+# Context to make it easier to profile bits of the code
+class profile:
```
Comment (jmeyers314) [inline galsim/utilities.py:2120]: FWIW, I'd actually love to rename this one so it doesn't conflict with using `@profile` via [line_profiler](https://github.com/pyutils/line_profiler#line_profiler).

File: galsim/utilities.py [hunk 2]
```diff
@@ -1913,3 +1913,244 @@ def __exit__(self, type, value, traceback):
...
+    4. copy.copy(obj) returns an equivalent copy unless random=True.
+    5. copy.deepcopy(obj) returns an equivalent copy.
+    6. eval(repr(obj)) returns an equivalent copy unless random or irreprable=True.
+    7. repr(obj) makes something if irreprable=False.
+
+    Parameters:
+        obj:            The object to test
+        func:           A function of obj to use to test for equivalence.  [default: lambda x: x]
+        irreprable:     Whether to skip the eval(repr(obj)) test.  [default: False]
+        random:         Whether the obj has some intrinsic randomness. [default: False, unless
```
Comment (jmeyers314) [inline galsim/utilities.py:1939]: Looks like default is `None`?

**Review by jmeyers314 — APPROVED** [review]
LGTM. Minor comments.

File: galsim/utilities.py (hunk 2 above)
Comment (rmjarvis) [reply galsim/utilities.py:1939]: Right.  I was describing what that default means, which is False normally, but tries to notice randomness.

File: galsim/utilities.py (hunk 1 above)
Comment (rmjarvis) [reply galsim/utilities.py:2120]: OK.  Suggestions?  Maybe just capitalize?  `with Profile():`  That would fit in with most of our GalSim style for classes.

Comment (rmjarvis) [reply galsim/utilities.py:2120]: I was also thinking of renaming a couple others.  `do_pickle` and `all_diff` aren't great names either, tbh.  I was thinking 'check_pickle` and `check_all_diff` for those.

Comment (jmeyers314) [reply galsim/utilities.py]: Yeah, these all sound better to me.

**Pull request merged**